ckanext/azure_auth/auth_config.py

```diff
@@ -0,0 +1,248 @@
+            self.end_session_endpoint = openid_cfg['end_session_endpoint']
+            if config[ATTR_TENANT_ID] != 'adfs':
+                self.issuer = openid_cfg['issuer']
+            else:
+                self.issuer = openid_cfg['access_token_issuer']
+        except KeyError:
+            raise ConfigLoadErrorException
+        return True
+
+    def _load_federation_metadata(self):
+        server_url = config[ATTR_AD_SERVER]
+        base_url = '{}/{}'.format(server_url, config[ATTR_TENANT_ID])
+        if config[ATTR_TENANT_ID] == 'adfs':
+            adfs_config_url = (
+                f'{server_url}/FederationMetadata/2007-06/FederationMetadata.xml'
+            )
+        else:
+            adfs_config_url = (
+                f'{base_url}/FederationMetadata/2007-06/FederationMetadata.xml'
+            )
+
+        try:
+            log.info(f'Trying to get ADFS Metadata file {adfs_config_url}')
+            response = self.session.get(adfs_config_url, timeout=TIMEOUT)
+            response.raise_for_status()
+        except requests.HTTPError:
+            raise ConfigLoadErrorException
+
+        # Extract token signing certificates
+        xml_tree = ElementTree.fromstring(response.content)
+        cert_nodes = xml_tree.findall(XML_CERT_SECTIONS)
+        signing_certificates = [node.text for node in cert_nodes]
+
+        self._load_keys(signing_certificates)
+        self.issuer = xml_tree.get('entityID')
+        self.authorization_endpoint = f'{base_url}/oauth2/authorize'
+        self.token_endpoint = f'{base_url}/oauth2/token'
+        self.end_session_endpoint = f'{base_url}/ls/?wa=wsignout1.0'
+        return True
+
+    def _load_keys(self, certificates):
+        new_keys = []
+        for cert in certificates:
+            log.debug(f'Loading public key from certificate: {cert}')
+            cert_obj = load_der_x509_certificate(
+                base64.b64decode(cert), backend
+            )
+            new_keys.append(cert_obj.public_key())
+        self.signing_keys = new_keys
+
+    def build_authorization_endpoint(self):
+        '''
+        This function returns the ADFS authorization URL.
+
+        Args:
+            request(django.http.request.HttpRequest): A django Request object
+
+        Returns:
+            str: The redirect URI
+
+        '''
+        self.load_config()
+        redirect_to = '/'
+        redirect_to = base64.urlsafe_b64encode(redirect_to.encode()).decode()
+        query = {}
+        query.update(
+            {
+                'response_type': 'code',
+                'client_id': config[ATTR_CLIENT_ID],
+                'redirect_uri': config[ATTR_REDIRECT_URL],
+                'state': redirect_to,
+            }
+        )
+        if self._mode == 'openid_connect':
+            query['scope'] = 'openid'
+            if config[ATTR_DISABLE_SSO]:
+                query['prompt'] = 'login'
+            if config[ATTR_FORCE_MFA]:
```

Comment:
`config[ATTR_FORCE_MFA]` will return a str, which is evaluated as `true` even when its value is `"False"`.

Need to call `asbool` (from `ckan,common`): `asbool(config[ATTR_FORCE_MFA])`